libs/platform/src/lib/components/form/form-group/form-group.component.ts

```diff
@@ -282,56 +309,115 @@ export class FormGroupComponent implements FormGroupContainer, OnInit, AfterCont
-     * Since I want to have it cached I know zones ahead of time, but you can have this generic and
-     * store everything in Map. Since here we have OnPush CD strategy it should not be a big deal.
+     * @hidden
+     * Assign a fields or field group to specified columns with rank.
+     * if `columnLayoutType` is given a fields without column property will set on last column.
+     * Otherwise the fields set 1 column.
      */
-    private updateFieldByZone(): void {
-        const zLeft: Array<GroupField> = [];
-        const zRight: Array<GroupField> = [];
-
-        this.formFieldChildren.forEach((item, index) => {
-            const zone: FormZone = this._multiLayout ? item.zone || 'zLeft' : 'zLeft';
-            const field = new GroupField(zone, item.id, item.rank || index, item.renderer, item.columns, item.fluid);
-
-            switch (zone) {
-                case 'zTop':
-                    if (!this.tZone) {
-                        this.tZone = [];
-                    }
-                    field.columns = 12;
-                    this.tZone.push(field);
-                    break;
-                case 'zBottom':
-                    if (!this.bZone) {
-                        this.bZone = [];
+    private _updateFieldByColumn(): void {
+        const rows: { [key: number]: FieldColumn } | { [key: number]: FieldGroup } = {};
+        let rowNumber = 0;
+        let columns: FieldColumn = {};
+
+        for (const child of this.formChildren) {
+            if (this._isFieldChild(child)) {
+                const f = this._getField(child);
+                const columnNumber = this._validateFieldColumn(child.column);
+
+                if (!columns[columnNumber]) {
+                    columns[columnNumber] = [f];
+                } else {
+                    columns[columnNumber].push(f);
+                }
+            }
+
+            if (this._isFieldGroupChild(child)) {
+                const fieldGroupColumns: FieldColumn = {};
+                rows[rowNumber] = columns;
+                columns = {};
+                rowNumber++;
+
+                const group = child.fields.map(f => this._getField(f));
+                group.forEach(field => {
+                    const columnNumber = this._validateFieldColumn(field.column);
+
+                    if (!fieldGroupColumns[columnNumber]) {
+                        fieldGroupColumns[columnNumber] = [field];
+                    } else {
+                        fieldGroupColumns[columnNumber].push(field);
                     }
-                    field.columns = 12;
-                    this.bZone.push(field);
-                    break;
-                case 'zLeft':
-                    zLeft.push(field);
-                    break;
-                case 'zRight':
-                    zRight.push(field);
-                    break;
+                });
+
+                rows[rowNumber] = new FieldGroup(child.label, fieldGroupColumns);
+                rowNumber++;
```

Comment:
` rowNumber++;` twice in a code block? forgot to remove one?

Reply:
Still need. Added condition to avoid unnecessary increment